cmd/util/ledger/migrations/ordered_map_migration.go

```diff
@@ -0,0 +1,473 @@
+	return enc.CBOR.EncodeRawBytes(r)
+}
+
+func (r RawStorable) ByteSize() uint32 {
+	return uint32(len(r))
+}
+
+func (r RawStorable) StoredValue(storage atree.SlabStorage) (atree.Value, error) {
+	return r, nil
+}
+
+func (r RawStorable) ChildStorables() []atree.Storable {
+	return nil
+}
+
+func (r RawStorable) Storable(_ atree.SlabStorage, _ atree.Address, _ uint64) (atree.Storable, error) {
+	return r, nil
+}
+
+func splitPayloads(inp []ledger.Payload) (fvmPayloads []ledger.Payload, storagePayloads []ledger.Payload, slabPayloads []ledger.Payload) {
+	for _, p := range inp {
+		if state.IsFVMStateKey(
+			string(p.Key.KeyParts[0].Value),
+			string(p.Key.KeyParts[1].Value),
+			string(p.Key.KeyParts[2].Value),
+		) {
+			fvmPayloads = append(fvmPayloads, p)
+			continue
+		}
+		if bytes.HasPrefix(p.Key.KeyParts[2].Value, []byte(atree.LedgerBaseStorageSlabPrefix)) {
+			slabPayloads = append(slabPayloads, p)
+			continue
+		}
+		// otherwise this is a storage payload
+		storagePayloads = append(storagePayloads, p)
+	}
+	return
+}
+
+func (m *OrderedMapMigration) migrate(payload []ledger.Payload) ([]ledger.Payload, error) {
+	fvmPayloads, storagePayloads, slabPayloads := splitPayloads(payload)
+	if len(slabPayloads) != 0 {
+		return nil, fmt.Errorf(
+			"slab storages are not empty: found %d",
+			len(slabPayloads),
+		)
+	}
+
+	ledgerView := NewView(fvmPayloads)
+	m.initPersistentSlabStorage(ledgerView)
+	m.initIntepreter()
+
+	sortedByOwnerAndDomain := make(map[string](map[string][]Pair))
+
+	for _, p := range storagePayloads {
+		owner, entry :=
+			string(p.Key.KeyParts[0].Value),
+			string(p.Key.KeyParts[2].Value)
+		// if the entry doesn't contain a separator, we just ignore it
+		// since it is storage metadata and does not need migration
+		if !strings.Contains(entry, "\x1f") {
+			continue
+		}
+		splitEntry := strings.Split(entry, "\x1f")
+		domain, key := splitEntry[0], splitEntry[1]
+		value := p.Value
+
+		domainMap, domainOk := sortedByOwnerAndDomain[owner]
+		if !domainOk {
+			domainMap = make(map[string][]Pair)
+		}
+		keyValuePairs, orderedOk := domainMap[domain]
+		if !orderedOk {
+			keyValuePairs = make([]Pair, 0)
+		}
+		domainMap[domain] = append(keyValuePairs, Pair{Key: key, Value: value})
+		sortedByOwnerAndDomain[owner] = domainMap
+	}
+
+	for owner, domainMaps := range sortedByOwnerAndDomain {
+		for domain, keyValuePairs := range domainMaps {
+			for _, pair := range keyValuePairs {
+				storageMap := m.newStorage.GetStorageMap(common.MustBytesToAddress([]byte(owner)), domain)
+				// in the interest of not having to update Cadence and break its abstractions to allow
+				// this one-time migration, just use reflection to grab the unexported field
+				unsafeOrderedMap := reflect.ValueOf(storageMap).Elem().FieldByName("orderedMap")
+				orderedMap := reflect.NewAt(
+					unsafeOrderedMap.Type(),
+					unsafe.Pointer(unsafeOrderedMap.UnsafeAddr()),
+				).Elem().Interface().(*atree.OrderedMap)
+				orderedMap.Set(
```

Comment:
It would be safer not to use `RawStorable` as `atree.Value`.  `RawStorable.Storable()` returns the entire raw bytes.

For example, if `RawStorable` represents a Cadence array, `Storable()` needs to return array's storage ID instead because atree stores storage ID for mutable and very large immutable elements.